docker-images/kafka-connect/scripts/kafka_connect_config_generator.sh

```diff
@@ -42,9 +42,19 @@ if [ -n "$KAFKA_CONNECT_SASL_USERNAME" ] && [ -n "$KAFKA_CONNECT_SASL_PASSWORD_F
     else
         SECURITY_PROTOCOL="SASL_PLAINTEXT"
     fi
-
+    
     PASSWORD=$(cat /opt/kafka/connect-password/$KAFKA_CONNECT_SASL_PASSWORD_FILE)
-    SASL_MECHANISM="SCRAM-SHA-512"
+
+    if [ "x$KAFKA_CONNECT_SASL_MECHANISM" = "xplain" ]; then
+        SASL_MECHANISM="PLAIN"
```

Comment:
Apart from the `SASL_MECHSANISM`, you will need to set here also some new variable for the JAAS cofngiuration which would either use `org.apache.kafka.common.security.scram.ScramLoginModule` or the module for Plain.

Reply:
thanks for pointing out ... fixed in b3c885f